proxy/src/fully_qualified_authority.rs

```diff
@@ -41,27 +41,24 @@ impl FullyQualifiedAuthority {
             }
         };
 
-        // A fully qualified name ending with a dot is normalized by removing the
-        // dot and doing nothing else.
-        if name.ends_with('.') {
-            let authority = authority.clone().into_bytes();
-            let normalized = authority.slice(0, authority.len() - 1);
-            let authority = Authority::from_shared(normalized).unwrap();
-            let name = FullyQualifiedAuthority(authority);
-            return NamedAddress {
-                name,
-                use_destination_service: true,
-            }
-        }
-
         // parts should have a maximum 4 of pieces (name, namespace, svc, zone)
         let mut parts = name.splitn(4, '.');
 
         // `Authority` guarantees the name has at least one part.
         assert!(parts.next().is_some());
 
         // Rewrite "$name" -> "$name.$default_namespace".
-        let has_explicit_namespace = parts.next().is_some();
+        let has_explicit_namespace = match parts.next() {
```

Comment:
Not exactly tied to this change, but it would be nice if `fn normalize` had a comment describing the general logic being applied. Currently it says:

```
Normalizes the name according to Kubernetes service naming conventions
```

If there was a brief overview of the conventions, this would be slightly more approachable

Reply:
better yet: doctests?